.github/workflows/ci.yml

```diff
@@ -38,6 +38,9 @@ jobs:
       - name: Enable nightly-only tests
         run: echo RUSTFLAGS=${RUSTFLAGS}\ --cfg=thiserror_nightly_testing >> $GITHUB_ENV
         if: matrix.rust == 'nightly'
+      - name: Test no-default-features
+        run: cargo test --all --no-default-features
```

Comment:
This isn't enough to assert `no_std` compatibility as dependencies might still pull in `std`.
I would suggest adding something like:
```sh
cargo check --target aarch64-unknown-none --no-default-features
```

Reply:
There are no dependencies. This is the top level crate.

Reply:
Perfect, they are rare.

Reply:
Wait, I see
```rust
[dependencies]
proc-macro2 = "1.0.74"
quote = "1.0.35"
syn = "2.0.46"
```

Reply:
in a proc macro crate